tests/acls/acl.sh.in

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+BASICURL="http://127.0.0.1"
+BASICURLS="https://127.0.0.1"
+
+NETDATA_VARLIB_DIR="@varlibdir_POST@"
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+
+#change the previous acl file and with a new 
+#and store it on a new file
+change_file(){
+	sed "s/$1/$2/g" netdata.cfg > "$4"
+}
+
+change_ssl_file(){
+	KEYROW="ssl key = $3/key.pem"
+	CERTROW="ssl certificate = $3/cert.pem"
+	sed "s@ssl key =@$KEYROW@g" netdata.ssl.cfg > tmp
+	sed "s@ssl certificate =@$CERTROW@g" tmp > tmp2
+	sed "s/$1/$2/g" tmp2 > "$4"
+}
+
+run_acl_tests() {
+	#Give a time for netdata start properly
+	sleep 2
+
+	curl -v -k --tls-max 1.2 --create-dirs -o index.html "$2" 2> log_index.txt
+	curl -v -k --tls-max 1.2 --create-dirs -o netdata.txt "$2/netdata.conf" 2> log_nc.txt
+	curl -v -k --tls-max 1.2 --create-dirs -o badge.csv "$2/api/v1/badge.svg?chart=cpu.cpu0_interrupts" 2> log_badge.txt
+	curl -v -k --tls-max 1.2 --create-dirs -o info.txt "$2/api/v1/info" 2> log_info.txt
+	curl -H "X-Auth-Token: $1" -v -k --tls-max 1.2 --create-dirs -o health.csv "$2/api/v1/manage/health?cmd=LIST" 2> log_health.txt
+
+	TOT=$(grep -c "HTTP/1.1 200 OK" log_* | cut -d: -f2| grep -c 1)
+	TOT1=$(grep -c "HTTP/1.1 301" log_* | cut -d: -f2| grep -c 1)
+
+	if [ "$TOT" -ne "$3" ]; then
+		if [ "$TOT1" -eq "5" ]; then
+			echo -e "${GREEN}All the urls were redirected to https"
```

Comment:
I entered this setting:
```
bind to = *:19999=dashboard|registry|badges|management|netdata.conf *:20000=dashboard|registry|badges|management^SSL=optional *:20001=dashboard|registry|netdata.conf^SSL=force *:20002=dashboard|registry
```

I can see http://localhost:20001 just fine, I'm not getting redirected to https
I AM getting redirected if I go to http://localhost:20001/netdata.conf

Reply:
This was fixed with my last commit, but it is necessary to say something about how to run the tests:

- For everything works fine I had to compile with ./netdata-installer.sh, because the script is searching the Netdata software in the path.
- After to put netdata in the path(/usr/sbin/netdata in my case), I ran the script and everything worked fine.
- So I decided to access the browser, and I had problem with the browser, it did not redirect me initially as we hope. When I saw my /var/log/netdata/access.log I could see that the browser were not requesting everything, it was using cache. After I press "CTRL+SHIFT+R" it redirected the address "http://localhost:2001"

You will see that in the script I am copying and pasting the previous rules changing one to be optional and other to be forced. In the way that the code is written right now, the last will become the final rule for Netdata, as I described in a previous commentary in this PR. To separate for ports, I keep my suggestion, we will need another PR and we will need to define not policy for netdata, we will need to define SSL policy for ports.

Reply:
What is SSL ACL (access control list?)? This line?

> *:19999=dashboard|registry|badges|management|netdata.conf *:20000=dashboard|registry|badges|management^SSL=optional *:20001=dashboard|registry|netdata.conf^SSL=force *:20002=dashboard|registry

Reply:
The SSL on connections is defined in bind to along with others ACLs of the netdata, this is the motive I called it SSL ACL in the title, but case we wanna be rigorous with the terms, I agree SSL is not an ACL, it is a feature like the dbegnine.

Reply:
Let me explain the last changes, after our talk this morning where we clarified the whole situation, I changed the script to test the 4 possible scenarios:
- Without SSL
- With SSL optional
- With SSL force
- With both SSL options

For the two first scenarios we cannot have any kind of redirection.
For the last two, everytime that there is a request for a HTTP only, we must have a redirect. I am not showing more a message in green for this case, I decided to change for yellow, and the unique acceptable situation is redirect for all. I stopped the script to confirm that I was having 301 for all 5, and the results was as expected.
Another very important thing the previous recommendation continues valid, in other words, everything will work, if and only if, the binary that is generated by this PR is in the user path.

Reply:
It's still the wrong test. You always check for 5 redirections, when it's clear that with SSL=optional, I should NOT be redirecting, but accepting the http request.

Reply:
Hi @cakrit,

I changed the script for every time that it has the option SSL set in one port, it test both the SSL URL and the no encrypted URL, this time the tests are split and I am using a counter to be sure that we will have the 5 necessary redirects when the SSL is set to force. Unfortunately in my tests, I could not have what you described, I tested more than 10 times, and it redirects to 5, but now the script stops, so case you have the same problem, please send me the log_*.txt files for I check the logs and try to replicate here.
In the stream tests I tests the following scenarios:
1 - A port with SSL optional, and the streaming connection without SSL and with SSL.
2 - A port with SSL forced, and the streaming connection with SSL and giving the expected error without SSL.
3 - Various ports with the 3 possibilities(Without any SSL, with SSL as Optional and SSL as Force), for all the 3 cases I got the expected result.

Best regards!

Reply:
It's a logical issue. How can you ALWAYS have 5 redirects if you have a configuration that says SSL=optional? The requests to that port should NOT be redirecting. 

Look at this test you have:
```
change_ssl_file "$CONF" "    bind to = *=dashboard|registry|badges|management|netdata.conf *:20000=dashboard|registry|badges|management *:20001=dashboard|registry|netdata.conf^SSL=optional *:20002=dashboard|registry" "$CWD" "netdata.conf.test1"
...
run_acl_tests $TOKEN "$BASICURL:20001" 4
run_acl_tests $TOKEN "$BASICURLS:20001" 4
```
This test should be failing, with the code you have in `run_acl_tests`, because port 20001 has `SSL=optional`. The only reason it's not failing is because netdata is redirecting, though it should NOT be. So both the test and the netdata code are wrong.

Reply:
I explained for you the motives in our talk, so I am going direct to the solution, as you can see now the function has one more argument, so instead to verify log by log and count the number of errors, I am giving as argument the number of acceptable redirects for that specific request.